src/main/java/spoon/reflect/path/CtRole.java

```diff
@@ -127,4 +128,39 @@ public String getCamelCaseName() {
 	public String toString() {
 		return getCamelCaseName();
 	}
+
+	/**
+	 * @return the CtRole which is the real holder of this virtual CtRole or null if there is no super role.
+	 * 	For example {@link #TYPE_MEMBER} is super role of {@link #CONSTRUCTOR}, {@link #FIELD}, {@link #METHOD}, {@link #NESTED_TYPE}
+	 */
+	public CtRole getSuperRole() {
+		return superRole;
+	}
+
+	/**
+	 * @return sub roles of this super role or empty array if there is no sub role.
+	 * 	For example {@link #TYPE_MEMBER} is super role of {@link #CONSTRUCTOR}, {@link #FIELD}, {@link #METHOD}, {@link #NESTED_TYPE}
+	 *
+	 */
+	public CtRole[] getSubRoles() {
+		return subRoles;
+	}
+
+	private static final CtRole[] EMPTY = new CtRole[0];
+	private CtRole superRole;
+	private CtRole[] subRoles;
+
+	static {
+		for (CtRole role : values()) {
+			role.subRoles = EMPTY;
```

Comment:
Instead of doing this static loop, can't you make the assignment in a constructor of CtRole?

Reply:
I wanted to do that, but java compiler has problem with that.
```
private static final CtRole[] EMPTY = new CtRole[0];
private CtRole superRole = EMPTY; //<-- compilation error. 
```
I wonder why too..,.  but it is so.

Reply:
Some hints about why it does not work: https://stackoverflow.com/questions/29174345/illegal-reference-to-static-field-from-initializer

However it will work if you do not use a constant but directly ```private CtRole[] subRoles = new Ctrole[0];```.